From 50493eefd624319c3a0661bb257fbb3d13764ecc Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Wed, 10 Dec 2014 15:24:04 +0100
Subject: [PATCH 0232/2241] fix: thermal: add support for suspend/resume

	- There is no need to implement subroutine for suspend since there is no
	data to store before suspending
	- Bug fix for: SYSTEMSW-1411 'Thermal sensor does not work after resume
	from suspend'

Change-Id: Ibf3df264f1abdef3fd21d2c300d08a70cdd54b1f
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/15257
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/15366
Reviewed-on: http://vgitil04.il.marvell.com:8080/28001
Reviewed-by: Ofer Heifetz <oferh@marvell.com>
---
 drivers/thermal/armada_thermal.c | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/drivers/thermal/armada_thermal.c b/drivers/thermal/armada_thermal.c
index 5ee6dc4..c2d27d0 100644
--- a/drivers/thermal/armada_thermal.c
+++ b/drivers/thermal/armada_thermal.c
@@ -531,9 +531,23 @@ static int armada_thermal_exit(struct platform_device *pdev)
 	return 0;
 }
 
+static int armada_thermal_resume(struct platform_device *pdev)
+{
+	struct thermal_zone_device *thermal =
+		platform_get_drvdata(pdev);
+	struct armada_thermal_priv *priv = thermal->devdata;
+
+	priv->data->init_sensor(pdev, priv);
+
+	return 0;
+}
+
 static struct platform_driver armada_thermal_driver = {
 	.probe = armada_thermal_probe,
 	.remove = armada_thermal_exit,
+#ifdef CONFIG_PM
+	.resume = armada_thermal_resume,
+#endif
 	.driver = {
 		.name = "armada_thermal",
 		.of_match_table = armada_thermal_id_table,
-- 
2.7.4

